### PR TITLE
Add missing comments in YaruNavigationRailItem

### PR DESCRIPTION
**Pull request by Jupi007**
Imported from #399

## Pull request checklist

- [x] This PR does not introduce visual changes

### CODE REVIEW

File: lib/src/layouts/yaru_navigation_rail_item.dart [hunk 1]
```diff
@@ -27,11 +27,24 @@ class YaruNavigationRailItem extends StatefulWidget {
     required this.style,
   });
 
+  /// Whatever the related page item is selected in the rail.
```
Comment (jpnurmi) [inline lib/src/layouts/yaru_navigation_rail_item.dart:30]:
```suggestion
  /// Whether the related page item is selected in the rail.
```

File: lib/src/layouts/yaru_navigation_rail_item.dart [hunk 2]
```diff
@@ -27,11 +27,24 @@ class YaruNavigationRailItem extends StatefulWidget {
...
   final bool? selected;
+
+  /// Icon widget, displayed agaist the [label].
```
Comment (jpnurmi) [inline lib/src/layouts/yaru_navigation_rail_item.dart:33]:
```suggestion
  /// Icon widget, displayed beside the [label].
```

File: lib/src/layouts/yaru_navigation_rail_item.dart [hunk 3]
```diff
@@ -27,11 +27,24 @@ class YaruNavigationRailItem extends StatefulWidget {
...
   final Widget icon;
+
+  /// Label widget, displayed agaist the [icon].
```
Comment (jpnurmi) [inline lib/src/layouts/yaru_navigation_rail_item.dart:36]:
```suggestion
  /// Label widget, displayed beside the [icon].
```

File: lib/src/layouts/yaru_navigation_rail_item.dart [hunk 4]
```diff
@@ -27,11 +27,24 @@ class YaruNavigationRailItem extends StatefulWidget {
...
   final Widget label;
+
+  /// Optional string tooltip.
+  /// If not null a tooltip will be displayed on hover.
+  /// It should describe the whole tile if possible.
   final String? tooltip;
+
+  /// Callback called on tap the tile.
```
Comment (jpnurmi) [inline lib/src/layouts/yaru_navigation_rail_item.dart:44]:
```suggestion
  /// Callback called when the tile is tapped.
```

**Comment by Jupi007** [comment]
Thanks for the typography review @jpnurmi  👍

**Review by jpnurmi — APPROVED** [review]
LGTM even though I wouldn't trust my English proof-reading for anything more official :rofl:

**Pull request merged**